components/Slider/examples/SliderCompareExampleViewController.m

```diff
@@ -31,46 +37,57 @@ - (id)init {
+    self.typographyScheme = [[MDCTypographyScheme alloc] init];
   }
   return self;
 }
 
 - (void)viewDidLoad {
   [super viewDidLoad];
-  self.view.backgroundColor = [UIColor whiteColor];
+  self.view.backgroundColor = self.colorScheme.backgroundColor;
 
   // Load your Material Component here.
-  MDCSlider *slider = [[MDCSlider alloc] initWithFrame:CGRectMake(0, 0, 100, 27)];
-  slider.statefulAPIEnabled = YES;
-  [MDCSliderColorThemer applySemanticColorScheme:self.colorScheme toSlider:slider];
-  [slider addTarget:self
+  self.slider = [[MDCSlider alloc] initWithFrame:CGRectMake(0, 0, 100, 27)];
+  self.slider.statefulAPIEnabled = YES;
+  [MDCSliderColorThemer applySemanticColorScheme:self.colorScheme toSlider:self.slider];
+  [self.slider addTarget:self
                 action:@selector(didChangeMDCSliderValue:)
       forControlEvents:UIControlEventValueChanged];
-  [self.view addSubview:slider];
-  slider.center = CGPointMake(CGRectGetMidX(self.view.bounds), 46);
+  [self.view addSubview:self.slider];
+  self.label = [[UILabel alloc] init];
+  self.label.text = @"MDCSlider";
+  self.label.font = self.typographyScheme.body1;
 
-  UILabel *label = [[UILabel alloc] init];
-  label.text = @"MDCSlider";
-  [label sizeToFit];
-  [self.view addSubview:label];
-  label.center = CGPointMake(slider.center.x, slider.center.y + 2 * label.frame.size.height);
-  label.frame = MDCRectAlignToScale(label.frame, [UIScreen mainScreen].scale);
+  [self.label sizeToFit];
+  [self.view addSubview:self.label];
 
   // Vanilla  UISlider for comparison.
-  UISlider *uiSlider = [[UISlider alloc] initWithFrame:CGRectMake(0, 0, 100, 27)];
-  [uiSlider addTarget:self
+  self.uiSlider = [[UISlider alloc] initWithFrame:CGRectMake(0, 0, 100, 27)];
+  [self.uiSlider addTarget:self
                 action:@selector(didChangeUISliderValue:)
       forControlEvents:UIControlEventValueChanged];
-  [self.view addSubview:uiSlider];
-  uiSlider.center = CGPointMake(CGRectGetMidX(self.view.bounds), 184);
-
-  UILabel *uiSliderLabel = [[UILabel alloc] init];
-  uiSliderLabel.text = @"UISlider";
-  [uiSliderLabel sizeToFit];
-  [self.view addSubview:uiSliderLabel];
-  uiSliderLabel.center =
-      CGPointMake(uiSlider.center.x, uiSlider.center.y + 2 * uiSliderLabel.frame.size.height);
-  uiSliderLabel.frame = MDCRectAlignToScale(uiSliderLabel.frame, [UIScreen mainScreen].scale);
+  [self.view addSubview:self.uiSlider];
+
+  self.uiSliderLabel = [[UILabel alloc] init];
+  self.uiSliderLabel.text = @"UISlider";
+  self.uiSliderLabel.font = self.typographyScheme.body1;
+  [self.uiSliderLabel sizeToFit];
+  [self.view addSubview:self.uiSliderLabel];
+}
+
+- (void)viewWillLayoutSubviews {
+  [super viewWillLayoutSubviews];
+
+  self.slider.center = CGPointMake(CGRectGetMidX(self.view.bounds),
+                              CGRectGetMidY(self.view.bounds) - 50);
+  self.label.center = CGPointMake(self.slider.center.x, self.slider.center.y + 2 * self.label.frame.size.height);
```

Comment:
Please use CGRectGetHeight() for frame/bounds value accesses. Otherwise, please call CGRectStandardized and keep a local copy of the standardized rect.

Reply:
Done